src/transformers/commands/serving.py

```diff
@@ -390,39 +336,97 @@ def register_subcommand(parser: ArgumentParser):
-        # cache and avoid re-running prefil
-        self.last_messages = None
-        self.last_kv_cache = None
-
+        # Set up logging
         transformers_logger = logging.get_logger("transformers")
         transformers_logger.setLevel(logging.log_levels[self.args.log_level.lower()])
 
         cb_logger = logging.get_logger("transformers.generation.continuous_batching")
         cb_logger.setLevel(logging.log_levels[self.args.log_level.lower()])
 
-    def build_chat_completions_chunk(
+        # Internal state:
+        # 1. Tracks the most recently used model, to prevent reloading the model unnecessarily
+        self.loaded_model: Optional[str] = None
+        self.running_continuous_batching_manager: Optional[ContinuousBatchingManager] = None
+        self.model: PreTrainedModel
+        self.tokenizer: PreTrainedTokenizerFast
+
+        # 2. preserves information about the last call and last KV cache, to determine whether we can reuse the KV
+        # cache and avoid re-running prefil
+        self.last_messages = None
+        self.last_kv_cache = None
+
+    def validate_request(
+        self,
+        request: dict,
+        schema: "_TypedDictMeta",  # noqa: F821
+        validator: "TypeAdapter",
+        unused_fields: set,
+    ):
+        """
+        Validates the request against the schema, and checks for unexpected keys.
+
+        Args:
+            request (`dict`):
+                The request to validate.
+            schema (`_TypedDictMeta`):
+                The schema of the request to validate. It is a `TypedDict` definition.
+            validator (`TypeAdapter`):
+                The validator to use to validate the request. Built from `schema`.
+            unused_fields (`set`):
+                Fields accepted by `schema`, but not used in `transformers serve`.
+
+        Raises:
+            HTTPException: If the request is invalid or contains unexpected or unused fields.
+        """
+        logger.debug(f"Validating request: {request}")
+
+        # Validate unexpected keys -- Pydantic doesn't validate extra keys in the request.
+        input_keys = set(request.keys())
+        possible_keys = schema.__mutable_keys__
+        unexpected_keys = input_keys - possible_keys
+        if unexpected_keys:
+            logger.error(f"Unexpected keys in the request: {unexpected_keys}")
+            raise HTTPException(status_code=422, detail=f"Unexpected keys in the request: {unexpected_keys}")
+
+        if self.args.input_validation:
+            # Validate expected keys
+            try:
+                validator.validate_python(request)
+            except ValidationError as e:
+                logger.error(f"Validation error: {e.errors()}")
+                raise HTTPException(status_code=422, detail=e.errors())
+
+            # Validate unused fields
+            unused_fields_in_request = input_keys & unused_fields
```

Comment:
by default, we don't throw an exception if the request has a field for which we haven't implemented logic

(goal: fail silently for now, while we are figuring things out, and enable this in the future)